File: remote/src/main/java/com/kelsos/mbrc/services/api/LibraryService.java
package com.kelsos.mbrc.services.api;

import com.kelsos.mbrc.dto.PaginatedResponse;
import com.kelsos.mbrc.dto.library.AlbumDto;
import com.kelsos.mbrc.dto.library.ArtistDto;
import com.kelsos.mbrc.dto.library.CoverDto;
import com.kelsos.mbrc.dto.library.GenreDto;
import com.kelsos.mbrc.dto.library.TrackDto;
import retrofit2.http.GET;
import retrofit2.http.Query;
import rx.Observable;

public interface LibraryService {

  @GET("/library/genres") Observable<PaginatedResponse<GenreDto>> getLibraryGenres(@Query("offset") int offset,
      @Query("limit") int limit,
      @Query("after") long after);

  @GET("/library/tracks") Observable<PaginatedResponse<TrackDto>> getLibraryTracks(@Query("offset") int offset,
      @Query("limit") int limit,
      @Query("after") long after);

  @GET("/library/covers") Observable<PaginatedResponse<CoverDto>> getLibraryCovers(@Query("offset") int offset,
      @Query("limit") int limit,
      @Query("after") long after);

  @GET("/library/artists") Observable<PaginatedResponse<ArtistDto>> getLibraryArtists(@Query("offset") int offset,
      @Query("limit") int limit,
      @Query("after") long after);

  @GET("/library/albums") Observable<PaginatedResponse<AlbumDto>> getLibraryAlbums(@Query("offset") int offset,
      @Query("limit") int limit,
      @Query("after") long after);
}

File: mbrc-lib/src/main/kotlin/com/kelsos/mbrc/services/api/LibraryService.kt
package com.kelsos.mbrc.services.api

import com.kelsos.mbrc.dto.PaginatedResponse
import com.kelsos.mbrc.dto.library.*
import retrofit2.http.GET
import retrofit2.http.Query
import rx.Observable

interface LibraryService {

    @GET("/library/genres") fun getLibraryGenres(@Query("offset") offset: Int,
                                                 @Query("limit") limit: Int,
                                                 @Query("after") after: Long): Observable<PaginatedResponse<GenreDto>>

    @GET("/library/tracks") fun getLibraryTracks(@Query("offset") offset: Int,
                                                 @Query("limit") limit: Int,
                                                 @Query("after") after: Long): Observable<PaginatedResponse<TrackDto>>

    @GET("/library/covers") fun getLibraryCovers(@Query("offset") offset: Int,
                                                 @Query("limit") limit: Int,
                                                 @Query("after") after: Long): Observable<PaginatedResponse<CoverDto>>

    @GET("/library/artists") fun getLibraryArtists(@Query("offset") offset: Int,
                                                   @Query("limit") limit: Int,
                                                   @Query("after") after: Long): Observable<PaginatedResponse<ArtistDto>>

    @GET("/library/albums") fun getLibraryAlbums(@Query("offset") offset: Int,
                                                 @Query("limit") limit: Int,
                                                 @Query("after") after: Long): Observable<PaginatedResponse<AlbumDto>>
}
